pkg/scuzz/control.go

```diff
@@ -4,11 +4,43 @@
 
 package scuzz
 
-// Disk is the interface to a disk, with operations
-// to create packets and operate on them.
-type Disk interface {
-	// Unlock unlocks the drive, given a password
-	Unlock(string, uint, bool) error
-	// Identify returns drive identity information
-	Identify(timeout uint) error
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type (
+	// Info is information about a device.
+	Info struct {
+		NumberSectors    uint64
+		Cylinders        uint
+		Heads            uint
+		Sectors          uint
+		TrackSize        uint
+		SectorSize       uint
+		ECCBytes         uint
+		Serial           string
+		Model            string
+		FirmwareRevision string
+	}
+
+	// Disk is the interface to a disk, with operations
+	// to create packets and operate on them.
+	Disk interface {
```

Comment:
I've got a dumb question... but do you ever expect there to be an implementation of `Disk` that isn't `SGDisk`? Are there other Linux interfaces like this we'd implement it for?

Reply:
yes. I am hoping and hoping that at some point sysfs gets way better, and on bsd and darwin there is no sgdisk at all. It's not one of your all time great interfaces.